### PR TITLE
segmentation

### PR DESCRIPTION
**Pull request by zeppone**
ADD: segmentation.c segmention.h
CHANGE: :Makefile

### CODE REVIEW

File: src/load_Image/segmentation.c [hunk 1]
```diff
@@ -0,0 +1,252 @@
...
+/* return a list of int. That represent the ammout of black pixel of each line*/
+int * horizontalProjection(GdkPixbuf *pixbuf)
+{
+    guchar *p;
+    int rowstride = gdk_pixbuf_get_rowstride (pixbuf);
+    int n_channels = gdk_pixbuf_get_n_channels (pixbuf);
+    guchar *which_pixels = gdk_pixbuf_get_pixels (pixbuf);
+    int height = gdk_pixbuf_get_height (pixbuf);
+    int width = gdk_pixbuf_get_width (pixbuf);
+    int * hp =(int *) malloc( height );
```
Comment (JulesdeCube) [inline src/load_Image/segmentation.c]: add `sizeof(int)`

File: src/load_Image/segmentation.c [hunk 2]
```diff
@@ -0,0 +1,252 @@
...
+/* return a list of int. That represent the ammout of black pixel of each collum*/
+int * verticalProjection(GdkPixbuf *pixbuf)
+{
+    guchar *p;
+    int rowstride = gdk_pixbuf_get_rowstride (pixbuf);
+    int n_channels = gdk_pixbuf_get_n_channels (pixbuf);
+    guchar *which_pixels = gdk_pixbuf_get_pixels (pixbuf);
+    int height = gdk_pixbuf_get_height (pixbuf);
+    int width = gdk_pixbuf_get_width (pixbuf);
+    int * vp =(int *) malloc( width );
```
Comment (JulesdeCube) [inline src/load_Image/segmentation.c]: same as line 15

File: src/load_Image/segmentation.c [hunk 3]
```diff
@@ -0,0 +1,252 @@
...
+    int b = 0;
+    int height = gdk_pixbuf_get_height (pixbuf);
+    int width = gdk_pixbuf_get_width (pixbuf);
+    GdkPixbuf *pixLine;
+    int lineCount = 0;
+
+
+    for (int j = 0; j < height; ++j)
+    {
+        if (*(hp+j) == 0 )
```
Comment (JulesdeCube) [inline src/load_Image/segmentation.c]: `hp[j]` <=> `*(hp + j)`
`a == 0` <=> `!a`

File: src/load_Image/segmentation.c [hunk 4]
```diff
@@ -0,0 +1,252 @@
...
+            if (a != b)
+            {
+                lineCount +=1;
+                pixLine = gdk_pixbuf_new_subpixbuf(pixbuf,0,a,width,b-a);
+
+                /* line name is segmentedLine_/line number/ and the format is bmp*/
+                char * sg = (char *) malloc( 255 );
+                char * number = (char *) malloc( 255 );
+                char * t = (char *) malloc( 255 );
+                char * type = (char *) malloc( 255 );
```
Comment (JulesdeCube) [inline src/load_Image/segmentation.c]: add `sizeof`

File: src/load_Image/segmentation.c [hunk 5]
```diff
@@ -0,0 +1,252 @@
...
+    int width = gdk_pixbuf_get_width (pixbuf);
+    GdkPixbuf *pixChar;
+    int charCount = 0;
+
+
+
+    for (int j = 0; j < width; ++j)
+    {
+        //printf("black pixel in collum %d: %d\n",j,*(vp+j));
+        if (*(vp+j) == 0 )
```
Comment (JulesdeCube) [inline src/load_Image/segmentation.c]: same as line 84

File: src/load_Image/segmentation.c [hunk 6]
```diff
@@ -0,0 +1,252 @@
...
+                strcat(sg, number);
+                strcat(sg,type);
+                number +=1;
+                //printf("saving line\n");
+                gdk_pixbuf_save(pixLine,sg,t,NULL,NULL);
+
+
+
+                //moving file from UnRead to specific folder.
+                char newpath[255] = "./cuted_line/";
```
Comment (JulesdeCube) [inline src/load_Image/segmentation.c]: size can be smaller

File: src/load_Image/segmentation.h [hunk 7]
```diff
@@ -0,0 +1,26 @@
+#ifndef UNREAD_SEGMENTATION_H
```
Comment (JulesdeCube) [inline src/load_Image/segmentation.h:1]: to comment

**Review by JulesdeCube — CHANGES_REQUESTED** [review]
comment functions + remove or move to `test/` the image in `src/load_Image/Images`

**Review by JulesdeCube — COMMENTED** [review]
no change made

**Review by Winbay — DISMISSED** [review]

**Review by JulesdeCube — DISMISSED** [review]

**Review by JulesdeCube — APPROVED** [review]

**Pull request merged**